src/pixelator/pna/demux/process.py

```diff
@@ -249,52 +252,90 @@ def finalize_batched_groups(
     The sorting is done on the `marker_1` and `marker_2` columns and written to parquet metadata
     which allows for fast contiguous reads [marker_1, marker_2] groups.
 
-    :param work_dir: the path to the work directory containing the demuxed data
-    :returns: A list of paths to the Parquet files
-    :raises ValueError: If an invalid strategy is provided.
+    Params:
+        input_dir: the path to the work directory containing the demuxed data
+        output_dir: the path to the output directory
+        remove_intermediates: whether to remove the intermediate Arrow files after writing to parquet
+        strategy: the demultiplexing strategy to use. Can be "paired" or "independent"
+        memory: maximum amount of memory to use in bytes
     """
+    output_dir.mkdir(exist_ok=True, parents=True)
+
     if strategy == "independent":
-        return _finalize_batched_groups_independent(work_dir, remove_intermediates)
+        return _finalize_batched_groups_independent(
+            input_dir,
+            output_dir,
+            remove_intermediates=remove_intermediates,
+            memory=memory,
+        )
     elif strategy == "paired":
-        return _finalize_batched_groups_paired(work_dir, remove_intermediates)
+        return _finalize_batched_groups_paired(
+            input_dir,
+            output_dir,
+            remove_intermediates=remove_intermediates,
+            memory=memory,
+        )
     else:
         raise ValueError("Unknown strategy")
 
 
-def _finalize_batched_groups_paired(work_dir: Path, remove_intermediates: bool = True):
+def _finalize_batched_groups_paired(
+    input_dir: Path,
+    output_dir,
+    remove_intermediates: bool = True,
+    memory: int | None = None,
+):
+    """Post-process the demuxed data by sorting and writing to Parquet.
+
+    The demuxed data is written as intermediate parquet files in the work directory.
+    These are then sorted and deduplicated.
+
+    The sorting is done on the `marker_1` and `marker_2` columns and written to parquet metadata
+    which allows for fast contiguous reads of [marker_1, marker_2] groups.
+
+    Params:
+        input_dir:
+            the path to the work directory containing the demuxed data
+        output_dir:
+            the path to the output directory where the final parquet files are written
+        remove_intermediates:
+            Whether to remove the intermediate Arrow files after writing to parquet
```

Comment:
The intermediate files are `parquet` file now right? Just wanted to check - if so I think we should update the docs to reflect this.

Reply:
Yes, i changed this since duckdb does need a community maintained plugin to read arrow files that did not seem to work properly.